src/battle_ai_main.c

```diff
@@ -4901,12 +4735,238 @@ static s32 AI_CheckViability(u32 battlerAtk, u32 battlerDef, u32 move, s32 score
         break;
     } // move effect checks
 
+    // check move additional effects that are likely to happen
+    for (i = 0; i < gBattleMoves[move].numAdditionalEffects; i++)
```

Comment:
As discussed move this under an `EFFECT_HIT` case umbrella to restore previous AI viability checks. We can discuss potential changes after this PR is merged.